examples/discovery-search-app/src/App.js

```diff
@@ -85,9 +89,28 @@ function SearchPage() {
   );
 }
 
-function PreviewPage() {
+function PreviewPage(props) {
   const { setSelectedResult } = useContext(SearchApi);
 
+  function doRenderTab(testResult, content) {
```

Comment:
when looping through `tabs`, wouldn't `[]` still trigger an item in the array?

i feel like a similar approach would just be
```
const tabs = [
    {
      name: 'Document',
      Component: DocumentPreview
    }
];

if (canRenderCIDocument(props.document)) {
  tabs.push({ name: 'CI', Component: CIDocument });
}
```